Adding ACSPublicCoverage to Fairlearn (#1007)
fairlearn/fairlearn: docs/user_guide/datasets/acs_public_coverage.rst

> .. _acs-public-coverage:ACSPublicCoverage-------------------------Introduction^^^^^^^^^^^^^^^^^The ACSPublicCoverage dataset is one of five datasets created by Ding et al. [1]_ as an improved alternative to the popular UCI Adult dataset [2]_.Briefly, the UCI Adult dataset is commonly used as a benchmark dataset when comparingdifferent algorithmic fairness interventions. While UCI Adult is used to predict whether an individual's annual income is above $50,000, the authors created ACSPublicCoverage to predict whether an individual is covered by public health insurance.The authors compiled data from the American Community Survey (ACS) Public Use Microdata Sample (PUMS). Note that this is a different source than the Annual Social and Economic Supplement (ASEC) of the Current Population Survey (CPS) used to construct the original UCI Adult dataset.Ding et al. [#0]_ filtered the data such that ACSPublicCoverage only includes individuals under 65 years old had an income of less than $30,000... _acspubliccoverage_dataset_description:Dataset Description^^^^^^^^^^^^^^^^^^^The authors provide 2018 data for all 50 states and Puerto Rico.The dataset contains 1,138,289 rows and 19 features, which we describe below:.. list-table::   :header-rows: 1   :widths: 7 30   :stub-columns: 1   *  - Column name      - Description   *  - AGEP      - Age as an integer from 0 to 99   *  - SCHL      - Educational attainment:         1. No schooling completed         2. Nursery school or preschool         3. Kindergarten         4. Grade 1         5. Grade 2         6. Grade 3         7. Grade 4         8. Grade 5         9. Grade 6         10. Grade 7         11. Grade 8         12. Grade 9         13. Grade 10         14. Grade 11         15. Grade 12 (no diploma)         16. Regular high school diploma         17. GED or alternative credential         18. Some college but less than 1 year         19. 1 or more years of college credit but no degree         20. Associate's degree         21. Bachelor's degree         22. Master's degree         23. Professional degree beyond a bachelor's degree         24. Doctorate degree   *  - MAR      - Marital status:         1. Married         2. Widowed         3. Divorced         4. Separated         5. Never married or under 15 years old   *  - SEX      - Sex code:         1. Male         2. Female   *  - DIS      - Disability recode:         1. With a disability         2. Without a disability   *  - ESP      - Employment status of parents:         1. Living with two parents: both parents in labor force         2. Living with two parents: Father only in labor force         3. Living with two parents: Mother only in labor force         4. Living with two parents: Neither parent in labor force         5. Living with father: Father in the labor force         6. Living with father: Father not in labor force         7. Living with mother: Mother in the labor force         8. Living with mother: Mother not in labor force   *  - CIT      - Citizenship status:         1. Born in the U.S.         2. Born in Puerto Rico, Guam, the U.S. Virgin Islands, or the Northern Marianas         3. Born abroad of American parent(s)         4. U.S. citizen by naturalization         5. Not a citizen of the U.S.   *  - MIG      - Mobility status (lived here 1 year ago):         1. Yes, same house (nonmovers)         2. No, outside US and Puerto Rico         3. No, different house in US or Puerto Rico            *  - MIL      - Military service:         1. Now on active duty         2. On active duty in the past, but not now         3. Only on active duty for training in Reserves/National Guard         4. Never served in the military            *  - ANC      - Ancestry recode:         1. Single         2. Multiple         3. Unclassified         4. Not reported         8. Suppressed for data year 2018 for select PUMAs   *  - NATIVITY      - Nativity:         1. Native         2. Foreign born            *  - DEAR      - Hearing difficulty:         1. Yes         2. No            *  - DEYE      - Vision difficulty:         1. Yes         2. No            *  - DREM      - Cognitive difficulty:         1. Yes         2. No            *  - PINCP      - Total annual income per person as an integer between -19997 and 4209995 US dollars. Loss of $19998 or more is coded as -19998. Income of $4209995 or more is coded as 4209995.   *  - ESR      - Employment status recode:         1. Civilian employed, at work         2. Civilian employed, with a job but not at work         3. Unemployed         4. Armed forces, at work         5. Armed forces, with a job but not at work         6. Not in labor force            *  - ST      - State code:         Please see data dictionary at `ACS PUMS documentation <https://www.census.gov/programs-surveys/acs/microdata/documentation.2018.html>`_ for the full list of state codes.     *  - FER      - Gave birth to child within the past 12 months:         1. Yes         2. No   *  - RAC1P      - Race code         1. White alone         2. Black or African American alone         3. American Indian alone         4. Alaska Native alone         5. American Indian and Alaska native tribes specified; or American Indian or Alaska Native, not specified and no other races         6. Asian alone         7. Native Hawaiian and Other Pacific Islander alone         8. Some Other Race alone         9. Two or More racesThe target label is given by PUBCOV, which can be used for a binary classification task... list-table::   :header-rows: 1   :widths: 7 30   :stub-columns: 1   *  - Column name      - Description   *  - PUBCOV      - Public health coverage, with PUBCOV == 1 if the individual has public health coverage, else 0.. topic:: References:  .. [1] Frances Ding, Moritz Hardt, John Miller, Ludwig Schmidt `"Retiring Adult: New Datasets for Fair Machine Learning" <https://arxiv.org/pdf/2108.04884.pdf>`_,      Advances in Neural Information Processing Systems 34, 2021.  .. [2] R. Kohavi and B. Becker. "UCI Adult Data Set." UCI Meachine Learning Repository, 5, 1996.

Reviewer: There are some issues with end-of-line encoding.

Author: Thank you for spotting it; it's been fixed!

> .. _acs-public-coverage:ACSPublicCoverage-------------------------Introduction^^^^^^^^^^^^^^^^^The ACSPublicCoverage dataset is one of five datasets created by Ding et al. [1]_ as an improved alternative to the popular UCI Adult dataset [2]_.Briefly, the UCI Adult dataset is commonly used as a benchmark dataset when comparingdifferent algorithmic fairness interventions. While UCI Adult is used to predict whether an individual's annual income is above $50,000, the authors created ACSPublicCoverage to predict whether an individual is covered by public health insurance.The authors compiled data from the American Community Survey (ACS) Public Use Microdata Sample (PUMS). Note that this is a different source than the Annual Social and Economic Supplement (ASEC) of the Current Population Survey (CPS) used to construct the original UCI Adult dataset.Ding et al. [#0]_ filtered the data such that ACSPublicCoverage only includes individuals under 65 years old had an income of less than $30,000... _acspubliccoverage_dataset_description:Dataset Description^^^^^^^^^^^^^^^^^^^The authors provide 2018 data for all 50 states and Puerto Rico.The dataset contains 1,138,289 rows and 19 features, which we describe below:.. list-table::   :header-rows: 1   :widths: 7 30   :stub-columns: 1   *  - Column name      - Description   *  - AGEP      - Age as an integer from 0 to 99   *  - SCHL      - Educational attainment:         1. No schooling completed         2. Nursery school or preschool         3. Kindergarten         4. Grade 1         5. Grade 2         6. Grade 3         7. Grade 4         8. Grade 5         9. Grade 6         10. Grade 7         11. Grade 8         12. Grade 9         13. Grade 10         14. Grade 11         15. Grade 12 (no diploma)         16. Regular high school diploma         17. GED or alternative credential         18. Some college but less than 1 year         19. 1 or more years of college credit but no degree         20. Associate's degree         21. Bachelor's degree         22. Master's degree         23. Professional degree beyond a bachelor's degree         24. Doctorate degree   *  - MAR      - Marital status:         1. Married         2. Widowed         3. Divorced         4. Separated         5. Never married or under 15 years old   *  - SEX      - Sex code:         1. Male         2. Female   *  - DIS      - Disability recode:         1. With a disability         2. Without a disability   *  - ESP      - Employment status of parents:         1. Living with two parents: both parents in labor force         2. Living with two parents: Father only in labor force         3. Living with two parents: Mother only in labor force         4. Living with two parents: Neither parent in labor force         5. Living with father: Father in the labor force         6. Living with father: Father not in labor force         7. Living with mother: Mother in the labor force         8. Living with mother: Mother not in labor force   *  - CIT      - Citizenship status:         1. Born in the U.S.         2. Born in Puerto Rico, Guam, the U.S. Virgin Islands, or the Northern Marianas         3. Born abroad of American parent(s)         4. U.S. citizen by naturalization         5. Not a citizen of the U.S.   *  - MIG      - Mobility status (lived here 1 year ago):         1. Yes, same house (nonmovers)         2. No, outside US and Puerto Rico         3. No, different house in US or Puerto Rico            *  - MIL      - Military service:         1. Now on active duty         2. On active duty in the past, but not now         3. Only on active duty for training in Reserves/National Guard         4. Never served in the military            *  - ANC      - Ancestry recode:         1. Single         2. Multiple         3. Unclassified         4. Not reported         8. Suppressed for data year 2018 for select PUMAs   *  - NATIVITY      - Nativity:         1. Native         2. Foreign born            *  - DEAR      - Hearing difficulty:         1. Yes         2. No            *  - DEYE      - Vision difficulty:         1. Yes         2. No            *  - DREM      - Cognitive difficulty:         1. Yes         2. No            *  - PINCP      - Total annual income per person as an integer between -19997 and 4209995 US dollars. Loss of $19998 or more is coded as -19998. Income of $4209995 or more is coded as 4209995.   *  - ESR      - Employment status recode:         1. Civilian employed, at work         2. Civilian employed, with a job but not at work         3. Unemployed         4. Armed forces, at work         5. Armed forces, with a job but not at work         6. Not in labor force            *  - ST      - State code:         Please see data dictionary at `ACS PUMS documentation <https://www.census.gov/programs-surveys/acs/microdata/documentation.2018.html>`_ for the full list of state codes.     *  - FER      - Gave birth to child within the past 12 months:         1. Yes         2. No   *  - RAC1P      - Race code         1. White alone         2. Black or African American alone         3. American Indian alone         4. Alaska Native alone         5. American Indian and Alaska native tribes specified; or American Indian or Alaska Native, not specified and no other races         6. Asian alone         7. Native Hawaiian and Other Pacific Islander alone         8. Some Other Race alone         9. Two or More racesThe target label is given by PUBCOV, which can be used for a binary classification task... list-table::   :header-rows: 1   :widths: 7 30   :stub-columns: 1   *  - Column name      - Description   *  - PUBCOV      - Public health coverage, with PUBCOV == 1 if the individual has public health coverage, else 0.. topic:: References:  .. [1] Frances Ding, Moritz Hardt, John Miller, Ludwig Schmidt `"Retiring Adult: New Datasets for Fair Machine Learning" <https://arxiv.org/pdf/2108.04884.pdf>`_,      Advances in Neural Information Processing Systems 34, 2021.  .. [2] R. Kohavi and B. Becker. "UCI Adult Data Set." UCI Meachine Learning Repository, 5, 1996.

Reviewer 2: the raw formatting of this file looks really strange, not sure how to fix that

> .. _acs-public-coverage:

Reviewer 3: My understanding was that tags for cross-referencing should use dashes, or at least that's what the documentation uses as an example. Please help me learn if I'm incorrect https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations

ACSPublicCoverage
-------------------------

Introduction
^^^^^^^^^^^^^^^^^

The ACSPublicCoverage dataset is one of five datasets created by Ding et al. [1]_ 
as an improved alternative to the popular UCI Adult dataset [2]_.
Briefly, the UCI Adult dataset is commonly used as a benchmark dataset when comparing
different algorithmic fairness interventions. While UCI Adult is used to predict 
whether an individual's annual income is above $50,000, the authors created 

> ACSPublicCoverage to predict whether an individual is covered by public health insurance.

Reviewer 4: Can you elaborate on what public health insurance means? Is that Medicare, Medicaid, or something else? If you find a good resource you can also link to that or cite it.

Author: Yes, public health insurance in the U.S. means either Medicare or Medicaid. People under age 65 may qualify for Medicare (which is normally reserved for people over age 65) if they have certain disabilities. I will elaborate on this in the documentation

The authors compiled data from the American Community Survey (ACS) Public Use Microdata Sample (PUMS). 
Note that this is a different source than the Annual Social and Economic Supplement (ASEC) 
of the Current Population Survey (CPS) used to construct the original UCI Adult dataset.

> Ding et al. [#0]_ filtered the data such that ACSPublicCoverage only includes individuals under 65 years old 
> had an income of less than $30,000.

Reviewer 4: ```suggestion
Ding et al. [#0]_ filtered the data such that ACSPublicCoverage only includes individuals under 65 years old 
that had an income of less than $30,000.
```
Also, any idea why $30k? Is that relevant to who can get public coverage?

Author: Upon further research, I actually am not sure. It seems like the income requirements differ by state, so the $30,000 cutoff seems rather arbitrary, but I could have missed something.

Reviewer 3: Oh that's interesting. Page 25 of their [paper](https://arxiv.org/pdf/2108.04884.pdf) just says that the filter was just that PINCP > $30,000.

> .. _acspubliccoverage_dataset_description:

Reviewer 4: ```suggestion
.. _acs_public_coverage_dataset_description:
```
just for consistency with the first one

   *  - FER
      - Gave birth to child within the past 12 months:
         1. Yes
         2. No

   *  - RAC1P
      - Race code
         1. White alone
         2. Black or African American alone
         3. American Indian alone
         4. Alaska Native alone
         5. American Indian and Alaska native tribes specified; or American Indian or Alaska Native, not specified and no other races
         6. Asian alone
         7. Native Hawaiian and Other Pacific Islander alone
         8. Some Other Race alone
         9. Two or More races

The target label is given by PUBCOV, which can be used for a binary classification task.

.. list-table::
   :header-rows: 1
   :widths: 7 30
   :stub-columns: 1

   *  - Column name
      - Description

   *  - PUBCOV
      - Public health coverage, with PUBCOV == 1 if the individual has public health coverage, else 0

.. topic:: References:

  .. [1] Frances Ding, Moritz Hardt, John Miller, Ludwig Schmidt `"Retiring Adult: New Datasets for Fair Machine Learning" <https://arxiv.org/pdf/2108.04884.pdf>`_,
      Advances in Neural Information Processing Systems 34, 2021.

>   .. [2] R. Kohavi and B. Becker. "UCI Adult Data Set." UCI Meachine Learning Repository, 5, 1996.

Reviewer 4: Did you copy this from somewhere? If so, it must have that typo there, too.
```suggestion
  .. [2] R. Kohavi and B. Becker. "UCI Adult Data Set." UCI Machine Learning Repository, 5, 1996.
```

Author: Ah, thanks for catching that! @kspieks Think the typo must also be in ACSIncome too then

Reviewer 3: This typo was in my PR at some point because my text editor wasn't checking spelling and I was typing quickly 😬 It's been fixed in my PR now

.. _acs_public_coverage:
ACSPublicCoverage
-------------------------

Introduction
^^^^^^^^^^^^

The ACSPublicCoverage dataset is one of five datasets created by Ding et al. [1]_ 
as an improved alternative to the popular UCI Adult dataset [2]_.

> Briefly, the UCI Adult dataset is commonly used as a benchmark dataset when comparing

Reviewer 3: Maybe I'm misunderstanding but I thought only ACSIncome was presented as a replacement for UCI Adult while the other datasets were just new datasets to evaluate algorithmic fairness (not direct improvements on UCI Adult?). The paper was titled Retiring Adult and thus only analyzed models trained on ACSIncome since the others have different prediction tasks and thus aren't really comparable? My initial thought is that describing UCI Adult isn't really relevant here, but I welcome other opinions :)

Author: That's a good point! I have clarified it as an improved fairness benchmark rather than a direct alternative. Thanks for pointing that out!

The target label is given by PUBCOV, which can be used for a binary
classification task.

.. list-table::
   :header-rows: 1
   :widths: 7 30
   :stub-columns: 1

   *  - Column name
      - Description

   *  - PUBCOV
      - Public health coverage, with PUBCOV == 1 if the individual has public
        health coverage, else 0

.. _acs_public_coverage_discussion:

Discussion
^^^^^^^^^^

A meaningful evaluation of algorithmic (un)fairness considers the context in
which the machine learning model will be used. In this way, we can avoid
falling into abstraction traps (see :footcite:`selbst2019fairness`). For
example, the relationship between demographics and healthcare coverage may vary
widely by state - a model that works well in one state may not be suitable to
use in a different state.

To investigate how viable ACSPublicCoverage is as a fairness benchmark dataset,
we must first ask what real-world tasks might require a prediction of whether
an individual is covered by public health insurance. Usually, the entity that
needs this information (e.g., the government checking for eligibility in a
social program) will simply ask the individual directly. Ding et al. do not

> provide any examples, so it is difficult to evaluate what notion of fairness is

Reviewer 5: Definitely agreed that this is difficult for the author of this documentation to do in the abstract, but is the implication also that this would be difficult for users of this dataset?

Reviewer 6: To be honest I am still very confused about the prediction task and I feel like we could maybe expand a bit on why it is important that the task makes sense before we can even talk about fairness.

It reminds me a bit about the "studying up" paper; why are we focussing on low-income individuals. How would something like this be used to help them? Or could this prediction task also be misused by, I don't know, predatory health insurance companies or "fraud" detection by the government (i.e., the government uses the model to check whether people are "honest" about whether they are covered by public health insurance)? I have no idea whether these examples make any sense, because I don't know much about the US context, but reading this paragraph did not clear anything up for me.

@mmadaio @MiroDudik @romanlutz being more familiar with the US context, do you have thoughts?

Reviewer 4: @britneyting @kspieks @imatiach-msft and I were discussing this back in January as well. I share your concerns. In fact, I feel the same about the income prediction and all the other tasks mentioned in the paper. The context in which this task is useful is not obvious to me. That said, I do believe that this is something we can discuss in the user guide. It leads to the bigger question of "Should benchmarks always be realistic?" I have thoughts on that... Can we perhaps push people to more realistic datasets by outlining shortcomings?

Anyway, curious to hear what you and others think, @hildeweerts .

Reviewer 7: we can also open an issue and discuss it further and not block this PR for this.

Reviewer 6: @romanlutz given that you have thought about this extensively, do you have a suggestion to rephrase this paragraph to make it clearer what our concerns are?

I like @adrinjalali's suggestion to open a new issue to discuss the more general problem more thoroughly.

Reviewer 4: I'll have to think about this a bit more to phrase it right. It may take me a while, though. 

I'm not sure whether we should proceed with this anyway (as we already have Adult and the other variation) and open an issue for describing this sort of problem, or block this PR instead. I see the concern (as mentioned earlier), but given that we have others with the same issue it feels like an odd point to block. On the other hand, that opens the door for more and more contributions that don't interrogate this problem (at least until we capture this in some way).

Reviewer 6: I am also not sure... if we block the PR now I don't think it will get merged in the foreseeable future (if ever), but if we do not touch upon this issue at all we also create more "technical" debt.

I think it should be possible to rephrase the current text a bit to make our concerns clearer, without interrogating the problem in detail. E.g., I think some examples can go a long way, e.g., who is using it (e.g., a governmental organization versus industry) and for what purpose (e.g., a model used to distribute benefits versus a model that is used for fraud detection) already makes a huge difference in how we think about fairness.

Reviewer 4: I agree 100% in the general case. 

In this particular one I'm still wondering what that might be. Any government org that needs to know your insurance status can just ask. Same for companies. And if they don't feel like they can ask they probably aren't meant to know and it's a nefarious or at least borderline/grey area activity. Besides, I just can't wrap my head around why one would have ALL THE OTHER features but just not this one? That seems like a significant amount of detail we know about these individuals, how come we know all that and still need to guess the insurance status? Again, if we lawfully knew this much for a good reason they'd probably tell us that status, too. It's entirely possible my reasoning is flawed somewhere, of course.

Reviewer 6: You raise very good points - especially because it is based on such an apparently well-defined concept, the need for a predictive model seems... odd. I'm inclined to agree with your former comment and say we should just not proceed with this at all.

It's a pity of all the work that went into this, but perhaps sunk cost is not the best argument for including the dataset.

applicable without further context. 

The remainder of this discussion focuses on preliminary data analysis of

> ACSPublicCoverage and identifies potential problems that users should be aware

Reviewer 5: Who are the "users" here, and what are they using it for? People (e.g., data scientists?) who might use this dataset for fairness benchmarks? That may be worth calling out explicitly.

Reviewer 6: I think the main audience would be either researchers who want to benchmark some fair-ml approach, or people who want to use the dataset as an educational resource for educators/blog post writers who are looking for a dataset to use in examples/tutorials/etc. 

I have attempted to rewrite the piece a bit, but I am not sure whether everything I have written is correct. It would be great if some of our US citizens could check whether my interpretation is correct.

```
Another potential issue that researchers and educators should be aware 
of before using this dataset as a benchmark or educational resource, is the
construct validity of the measurements (see also :ref:`construct_validity`).
The purpose of ACSPublicCoverage is to "predict whether a low-income 
individual, not eligible for Medicare, has coverage from public health 
insurance".
Here, Ding et al. define 'low-income' as individuals whose income is 
below $30.000.
This income threshold `was calculated
<https://github.com/zykls/folktables/issues/14>`_ 
approximately in accordance with Affordable Care Act (ACA) eligibility 
guidelines, qualifying adults with income up to 133% of the 2021 Federal 
Poverty Line for Medicaid.
An important question is to what extent the $30.000 threshold is a valid
measurement of low income, in relation to public health insurance 
eligibility.
Firstly, the dataset contains data from 2018, in which the federal poverty 
line was slightly different compared to 2021.
More importantly, however, the federal poverty line is the same across
all states, whereas the low-income threshold used to determine 
eligibility for Medicaid can differ per state.
In particular, not all states expanded Medicaid under the ACA. 
Consequently, some states have a much lower low-income threshold 
compared to the federal poverty line.
Additionally, the dataset concerns individuals, but low-income status also
depends on household size, which is not one of the 19 features in 
ACSPublicCoverage.
We observe a similar issue with the eligibility for Medicare, which 
is implemented through the exclusion of individuals of 65 years and older.
While 65 traditionally is the eligibilty age for Medicare, younger 
people with disabilities may still qualify :footcite:`usgov2014who`.
Depending on how the predictions are used, the mismeasurement of 
low income and Medicare eligibility may result in allocation harms 
based on state or disability status (see also :ref:`types_of_harms`).
```

From this page, it looks like Medicaid eligibility may differ per state: https://www.healthcare.gov/lower-costs/

    prediction tasks that fall in that category. Ding et al. filters ACS PUMS
    for all individuals whose income is below $30,000, but the 'low-income'
    threshold differs for every state. Consequently, applying one threshold to
    all states can lead to flawed results. The paper does not justify the
    choice of $30,000, but the authors answer in `this Github issue
    <https://github.com/zykls/folktables/issues/14>`_ that it was calculated
    approximately in accordance with Affordable Care Act (ACA) eligibilty
    guidelines qualifying adults with income up to 133% of the 2021 Federal
    Poverty Line for Medicaid. Note that we examine the 2018 dataset in this
    article, and 2018 had a different poverty line. Furthermore, not all states
    expanded Medicaid under the ACA, so some states have a much lower poverty
    line.

    The authors acknowledge that the calculation was a very rough
    approximation, and ACSPublicCoverage is not designed to accommodate all the
    subtleties that contribute to the low-income designation and subsequent
    Medicaid eligibility. As a result, users of ACSPublicCoverage should be

>     aware of the dataset's limitations and possibly modify the data for their

Reviewer 5: By modify the data, do you mean the low-income threshold, specifically? Or other features too

    task.

2. Slight misstatement of data contents.

    The paper states that ACSPublicCoverage excludes individuals not eligible
    for Medicare, but this is not strictly true. While 65 traditionally is the
    eligibilty age for Medicare, and the data filters for people under 65 years

>     old, younger people with disabilities may still qualify

Reviewer 5: This may be a helpful nuance to communicate earlier - lines 14-15 are a bit confusing on this point.

.. _acs_public_coverage:
ACSPublicCoverage
-------------------------

Introduction
^^^^^^^^^^^^

The ACSPublicCoverage dataset is one of five datasets created by Ding et al.

> [1]_ as an improved fairness benchmark dataset to the popular UCI Adult dataset

Reviewer 7: done, hope they're okay.

         
   *  - ST
      - State code: Please see data dictionary at `ACS PUMS documentation
         <https://www.census.gov/programs-surveys/acs/microdata/documentation.2018.html>`_
         for the full list of state codes.

   *  - FER
      - Gave birth to child within the past 12 months:
         1. Yes
         2. No

   *  - RAC1P
      - Race code
         1. White alone
         2. Black or African American alone
         3. American Indian alone
         4. Alaska Native alone
         5. American Indian and Alaska native tribes specified; or American
            Indian or Alaska Native, not specified and no other races
         6. Asian alone
         7. Native Hawaiian and Other Pacific Islander alone
         8. Some Other Race alone
         9. Two or More races

The target label is given by PUBCOV, which can be used for a binary
classification task.

.. list-table::
   :header-rows: 1
   :widths: 7 30
   :stub-columns: 1

   *  - Column name
      - Description

   *  - PUBCOV
      - Public health coverage, with PUBCOV == 1 if the individual has public
        health coverage, else 0

.. _acs_public_coverage_discussion:

Discussion
^^^^^^^^^^

> As Ding et al. [1]_ notes, algorithmic (un)fairness must be evaluated in

Reviewer 6: This paragraph is a bit vague. If with "intervention" we refer to an "algorithmic fairness intervention", it is unclear what is meant with the "context of the intervention". How about something like:

```suggestion
A meaningful evaluation of algorithmic (un)fairness considers the context in which the machine learning model will be used. In this way, we can avoid falling into abstraction traps (see :ref:`abstraction_traps`). For example, the relationship between demographics and healthcare coverage may vary widely by state - a model that works well in one state may not be suitable to use in a different state. 

To investigate how viable ACSPublicCoverage is as a fairness benchmark dataset, we must first ask what real-world tasks might require a prediction of whether an individual is covered by public health insurance.
```

Reviewer 7: not sure about the reference, please check

context of the intervention, and the effects of different interventions vary
widely by state. Similarly, datasets should be carefully designed for the
specific prediction task in mind.

Therefore, to investigate how viable ACSPublicCoverage is as a fairness
benchmark dataset, we must first ask what real-world tasks might require a
prediction of whether an individual is covered by public health insurance.

> Usually, the entity that needs this information (e.g. the government checking

Reviewer 6: ```suggestion
Usually, the entity that needs this information (e.g., the government checking
```

> for eligibilty in a social program) will simply ask the individual directly.

Reviewer 6: I would expect that eligibility is based on a predetermined policy, does the government need to ask individuals directly or will they just determine eligibility based on provided information?

Reviewer 7: do we have to clarify?

Ding et al. do not provide any examples, so it is difficult to evaluate what

> "fair" even means without context. For now, the rest of this discussion focuses

Reviewer 6: ```suggestion
notion of fairness is applicable without further context. 

The remainder of this discussion focuses
```

on preliminary data analysis of ACSPublicCoverage and identifies potential
problems that users should be aware of beforehand.

1. A uniform $30,000 threshold ignores differences between states.

    Although the purpose of ACSPublicCoverage is to "predict whether a
    low-income individual, not eligible for Medicare, has coverage from public
    health insurance," this dataset should not be blindly used for all

>     prediction tasks that fall in that category. Ding et al. filters ACS PUMS

Reviewer 6: what category?

Reviewer 7: how would I know :D

>     for all individuals whose income is below $30,000, but the 'low-income'

Reviewer 6: What is meant with 'low-income' threshold here? Is this a specific threshold that is set by the government or are we referring more generally to the fact that what is "low" depends on the distribution of income, which can vary across states?

Reviewer 5: So, the US Census Bureau does have a definition of a [poverty line](https://www.census.gov/topics/income-poverty/poverty/guidance/poverty-measures.html), but it's calculated at the household level, not individually.

>     threshold differs for every state, so applying one threshold to all states

Reviewer 6: ```suggestion
    threshold differs for every state. Consequently, applying one threshold to all states
```

>     can lead to flawed results. The paper does not justify the choice of

Reviewer 6: Can we be more specific what the consequences would be?

    $30,000, but the authors answer in `this Github issue
    <https://github.com/zykls/folktables/issues/14>`_ that it was calculated
    approximately in accordance with Affordable Care Act (ACA) eligibilty
    guidelines qualifying adults with income up to 133% of the 2021 Federal
    Poverty Line for Medicaid. Note that we examine the 2018 dataset in this
    article, and 2018 had a different poverty line. Furthermore, not all states

>     expanded Medicaid under the ACA, so some states have a much lower poverty

Reviewer 6: Is this poverty line what is meant with the "low-income threshold" mentioned earlier?

Reviewer 5: Hmm, my understanding (from [here](https://www.census.gov/topics/income-poverty/poverty/about.html)) is that there _isn't_ a state-level poverty line, only a federal one, and it's updated annually to keep track with inflation via the CPI... but this may not be what the authors are referring to.

    line.

    The authors acknowledge that the calculation was a very rough
    approximation, and ACSPublicCoverage is not designed to accommodate all the
    subtleties that contribute to the low-income designation and subsequent
    Medicaid eligibility. As a result, users of ACSPublicCoverage should be
    aware of the dataset's limitations and possibly modify the data for their
    task.

2. Slight misstatement of data contents.

    The paper states that ACSPublicCoverage excludes individuals not eligible
    for Medicare, but this is not strictly true. While 65 traditionally is the
    eligibilty age for Medicare, and the data filters for people under 65 years
    old, younger people with disabilities may still qualify [3]_.

3. Missing features that contribute to low-income status.

    The dataset focuses on individuals, but low-income status also depends on
    household size, which is not one of the 19 features in ACSPublicCoverage.
    This could affect prediction tasks.

> 

Reviewer 6: Although all these points (1,2,3) are valid critiques of the dataset, the current phrasing feels a bit like we are nitpicking the authors, rather than providing valuable insights for people who are learning about fairness or consider using the data set. 

I think point (1) is very important, as this questions the validity of the task as a whole: is a single threshold of 30.000 a valid proxy for low income? We can make a connection to some of our text on construct validity here, I think. 

The other two points might be better off as some final small remarks.

Reviewer 7: cc @MiroDudik

Reviewer 5: Agreed with @hildeweerts about the framing, but I also think point 3 belongs with point 1, in a larger point about the construct validity of "low-income," given the state-level differences, and that it's calculated at a household level, not individually.
